docs/source/tuning.rst

```diff
@@ -0,0 +1,139 @@
+.. Copyright 2016 The Cartographer Authors
+
+.. Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+..      http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+.. cartographer SHA: ea7c39b6f078c693b92fed06d86ca501021147d9
+.. cartographer_ros SHA: 44459e18102305745c56f92549b87d8e91f434fe
+
+Tuning
+======
+
+Tuning Cartographer is unfortunately really difficult.
+The system has many parameters many of which affect each other.
+This tuning guide tries to explain a principled approach on concrete examples.
+
+Two systems
+-----------
+
+Cartographer can be seen as two separate, but related systems.
+The first one is local SLAM (sometimes also called frontend).
+Its job is build a locally consistent set of submaps and tie them together, but it will drift over time.
+Most of its options can be found in `trajectory_builder_2d.lua`_ for 2D and `trajectory_builder_3d.lua`_ for 3D.
+
+.. _trajectory_builder_2d.lua: https://github.com/googlecartographer/cartographer/blob/ea7c39b6f078c693b92fed06d86ca501021147d9/configuration_files/trajectory_builder_2d.lua
+.. _trajectory_builder_3d.lua: https://github.com/googlecartographer/cartographer/blob/ea7c39b6f078c693b92fed06d86ca501021147d9/configuration_files/trajectory_builder_3d.lua
+
+The other system is global SLAM (sometimes called the backend).
+It runs in background threads and its main job is to find loop closure constraints.
+It does that by exhaustively scan matching scans against submaps.
```

Comment:
Let's not call it "exhaustive" since we do not enumerate but rather use branch-and-bound.